egui_extras/src/layout.rs

```diff
@@ -0,0 +1,163 @@
+use egui::{Pos2, Rect, Response, Rgba, Sense, Ui};
+
+#[derive(Clone, Copy)]
+pub(crate) enum CellSize {
+    /// Absolute size in points
+    Absolute(f32),
+    /// Take all available space
+    Remainder,
+}
+
+/// Cells are positioned in two dimensions, cells go in one direction and form lines.
+///
+/// In a strip there's only one line which goes in the direction of the strip:
+///
+/// In a horizontal strip, a `[Layout]` with horizontal `[CellDirection]` is used.
+/// Its cells go from left to right inside this `[Layout]`.
+///
+/// In a table there's a `[Layout]` for each table row with a horizonal `[CellDirection]`.
+/// Its cells go from left to right. And the lines go from top to bottom.
+pub(crate) enum CellDirection {
+    /// Cells go from left to right
+    Horizontal,
+    /// Cells go fromtop to bottom
+    Vertical,
+}
+
+/// Positions cells in `[CellDirection]` and starts a new line on `[Layout::end_line]`
+pub struct Layout<'l> {
```

Comment:
Another potential name clash problem: there is already an `egui::Layout`, though it is not `pub`.

`Layout` is also a very general name. Perhaps `StripLayout` or something would work?